packages/snap-controller/src/Abstract/AbstractController.ts

```diff
@@ -66,12 +66,19 @@ export abstract class AbstractController {
 			} = event;
 
 			const beaconPayload: TrackErrorEvent = {
-				filename: filename || `${this.id} (${this.type.charAt(0).toUpperCase() + this.type.slice(1)}Controller)`,
+				filename,
 				stack,
 				message,
 				colno,
 				lineno,
 				errortimestamp: timeStamp,
+				details: additionalDetails ? JSON.stringify(additionalDetails) : undefined,
```

Comment:
As mentioned above (and against what was discussed elsewhere) - I don't think we need to do this stringify here. We should test.